includes/derivatives.inc

```diff
@@ -315,13 +319,15 @@ function islandora_pdf_add_jpg_derivative(AbstractObject $object, $file_uri, $ds
  *   The width to make the derived datastream.
  * @param int $height
  *   The height to make the derived datastream.
+ * @param int $colorspace
+ *   The colorspace for the derived datastream.
  *
  * @return string|array
  *   A URI to the generated derivative if successful, error message otherwise.
  *
  * @see hook_islandora_derivative()
  */
-function islandora_pdf_create_jpg_derivative($file_uri, $dsid, $width, $height) {
+function islandora_pdf_create_jpg_derivative($file_uri, $dsid, $width, $height, $colorspace) {
```

Comment:
This function sees [quite a bit of use in the wild](https://github.com/search?q=islandora_pdf_create_jpg_derivative&type=Code) and the change in signature would necessitate a deprecation cycle.

If I can make a recommendation ... perhaps we could pass in `NULL` here by default and use that to default to `RGB` while popping the deprecation warning; something to the effect of:

```php
function islandora_pdf_create_jpg_derivative($file_uri, $dsid, $width, $height, $colorspace = NULL) {
  if (is_null($colorspace)) {
    $colorspace = 'RGB';
    $message = islandora_deprecated('7.x-1.12', t('The colorspace now needs to be specified in islandora_pdf_create_jpg_derivative'));
    trigger_error(filter_xss($message), E_USER_DEPRECATED);
  }
...
```

Reply:
Agreed 👍